### DISCUSSION TITLE
Worker ruby processes are not terminated even service is stopped from Windows Service console

### DISCUSSION THREAD

**Discussion by deepaksb2** [fluent/fluentd#4373] [Q&A]
In one of our environment we observed very strange issue, we stopped the fluentd service from Services console and it stopped successfully but few worker thread processes are still not terminated, we tried to kill those worker processes but not getting terminated even forcefully (while service gets stopped)
 Has anyone faced the similar issue with windows td-agent? 

OS version: Windows Server 2016.
- Fluentd version: 1.16.2
- TD Agent version:  4.5.1

**Comment by daipom** [comment]
Depending on the plugins you are using, it may be possible that some worker processes can not be terminated.
It is because the stopping processes of each plugin are called when the worker process stops. 

On the other hand, the following cases have been reported as overall problems with Fluentd.

* #3937 

On Windows, repeating service restart at high speed can cause orphan supervisor/worker processes.

> we tried to kill those worker processes but not getting terminated even forcefully

Does this mean that you can't kill the processes?

**Reply by deepaksb2** [reply]
Yes, the system does not allow to kill the process , it reports access is denied, the issue persist until system is rebooted.
